src/firefly/js/visualize/task/LSSTFootprintTask.js

```diff
@@ -0,0 +1,354 @@
+                        dispatchDestroyDrawLayer} from '../DrawLayerCntlr.js';
+import {logError} from '../../util/WebUtil.js';
+import {ImageLineBasedObj} from '../draw/ImageLineBasedObj.js';
+import ImagePlotCntlr, {visRoot, dispatchPlotImage} from '../ImagePlotCntlr.js';
+import {makeTblRequest, cloneRequest, MAX_ROW} from '../../tables/TableRequestUtil.js';
+import {dispatchAddActionWatcher} from '../../core/MasterSaga.js';
+import {getAViewFromMultiView, getMultiViewRoot, IMAGE} from '../MultiViewCntlr.js';
+import WebPlotRequest from '../WebPlotRequest.js';
+import {dispatchTableSearch, TABLE_LOADED, TABLE_SELECT,TABLE_HIGHLIGHT,TABLE_REMOVE,TABLE_UPDATE}
+                from '../../tables/TablesCntlr.js';
+import {getTblById, doFetchTable, getColumnIdx, getColumn} from '../../tables/TableUtil.js';
+import LSSTFootprint from '../../drawingLayers/ImageLineBasedFootprint';
+import {convertAngle} from '../VisUtil.js';
+
+
+export const isLsstFootprintTable = (tableModel, fromAnalysis = false, tbl_idx = 0) => {
+    const HEADER_KEY_COL = 1;
+    const HEADER_VAL_COL = 2;
+    const lsstKeys = ['contains_lsst_footprints', 'contains_lsst_measurements'];
+    const {tableMeta} = tableModel || {};
+
+    if (fromAnalysis && tableMeta) {      // check meta from analysis
+        const metaInfo = JSON.parse(get(tableMeta, [tbl_idx]));
+        const metaAry = metaInfo && get(metaInfo, ['tableData', 'data']);
+
+        if (metaAry && isArray(metaAry)) {
+            const contains = metaAry.reduce((prev, oneMeta) => {
+                if (prev.length === lsstKeys.length) return prev;  // all keys are set
+                if ((oneMeta.length > HEADER_VAL_COL) && lsstKeys.includes(oneMeta[HEADER_KEY_COL]) &&
+                     oneMeta[HEADER_VAL_COL] === 'true' && (!prev.includes(oneMeta[HEADER_KEY_COL]))) {
+                    prev.push(oneMeta[HEADER_VAL_COL]);
+                }
+                return prev;
+            }, []);
+
+            return (contains.length === lsstKeys.length);
+        }
+    } else if (tableMeta) {                                        // not from analysis summary table
+        const filterRes = Object.keys(tableMeta).reduce((prev, oneKey) => {
+            if (prev.length === lsstKeys.length) return prev;
+            if (lsstKeys.includes(oneKey) && tableMeta[oneKey] === 'true' && !prev.includes(oneKey)) {
+                prev.push(oneKey);
+            }
+            return prev;
+        }, []);
+        return (filterRes.length === lsstKeys.length);
+    }
+    return false;
+};
+
+// columns in footprint table
+const spans = 'spans';
+const peaks = 'peaks';
+const corner1_x = 'footprint_corner1_x';
+const corner1_y = 'footprint_corner1_y';
+const corner2_x = 'footprint_corner2_x';
+const corner2_y = 'footprint_corner2_y';
+const footprintid = 'id';
+const table_rowidx = 'ROW_IDX';
+const ra_col = 'coord_ra';
+const dec_col = 'coord_dec';
```

Comment:
I think we've made an assumption that the catalog containing footprints will have the coordinate columns in it. I am becoming worried that there will be many cases where the coordinate columns are not included. I have one such case from the August deblending workshop.

Reply:
current tableMeta has the property of 'CatalogCoordColumns'. it will be good to have other column names set in the tableMeta.

Reply:
yes, the block of hard code will be used as a fallback, also discussed with @stargaser, he will prepare the sample votable with meta data for 'CatalogCoordColumns', 'FootPrintColumnNames' and 'pixelsys'. The value of  'CatalogCoordColumns' will also indicate if there is columns with valid catalog coordinate data. Pixel coordinates will be used in case the catalog coordinate columns have no valid data.